src/InitControl.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2015 - 2023, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "config.h"
+
+#include "InitControl.hpp"
+
+#include <iostream>
+#include <stdexcept>
+#include <regex>
+
+#include "geopm/Exception.hpp"
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+
+#include "PlatformIOProf.hpp"
+
+namespace geopm
+{
+    std::unique_ptr<InitControl> InitControl::make_unique(const std::string &input_file)
+    {
+        return geopm::make_unique<InitControlImp>(input_file);
+    }
+
+    InitControlImp::InitControlImp(const std::string &input_file)
+        : InitControlImp(input_file,
+                         PlatformIOProf::platform_io())
+    {
+
+    }
+
+    InitControlImp::InitControlImp(const std::string &input_file,
+                                   PlatformIO &platform_io)
+        : m_platform_io(platform_io)
+    {
+        if (input_file != "") {
+            parse_input(input_file);
+        }
+    }
+
+    void InitControlImp::parse_input(const std::string &input_file)
+    {
+        std::string file_data = read_file(input_file);
+        std::vector<std::string> lines = string_split(file_data, "\n");
+
+        // (      Start capture group
+        // [+-]?  Optional + or -
+        // [0-9]* Zero or more digits
+        // [.]?   Optional .
+        // [0-9]+ One or more digits
+        // (?:    Begin non-capture group
+        // [eE]?  Optional e or E
+        // [+-]?  Optional + or -
+        // [0-9]+ One or more digits
+        // )?     End non-capture group, optional
+        // )      End capture group
+        std::regex sci_notation_regex("([+-]?[0-9]*[.]?[0-9]+(?:[eE]?[+-]?[0-9]+)?)");
+
+        // ^\\s*  String begins with zero or more whitespace characters
+        // (\\w+) Capture group #1 for one or more alphanumeric characters (CONTROL NAME)
+        // \\s+   Zero or more whitespace characters
```

Comment:
```suggestion
        // \\s+   One or more whitespace characters
```
Same for the others below